core/src/main/java/org/elasticsearch/index/translog/Translog.java

```diff
@@ -443,6 +456,20 @@ public Location add(final Operation operation) throws IOException {
     }
 
     /**
+     * Tests whether or not the current generation of the translog should be folded into a new
+     * generation. This test is based on the size of the current generation compared to the
+     * configured generation threshold size.
+     *
+     * @param translog the translog
+     * @return {@code true} if the current generation should be folded into a new generation
+     */
+    private static boolean shouldFoldGeneration(final Translog translog) {
```

Comment:
I'd wouldn't have declared this as static. I'm really curious why you did. I expect I have something to learn from this....